clio-server/src/main/scala/org/broadinstitute/clio/server/model/data.scala

```diff
@@ -2,9 +2,18 @@ package org.broadinstitute.clio.server.model
 
 import java.time.OffsetDateTime
 
+object ModelReadGroupLocation {
+  sealed abstract class Place(override val toString: String)
+  case object Gcp extends Place("GCP")
+  case object OnPrem extends Place("OnPrem")
+  val key = "location"
+  val unknown = (key, "unknown")
+}
+
 case class ModelReadGroupKey(flowcellBarcode: String,
                              lane: Int,
-                             libraryName: String)
+                             libraryName: String,
+                             location: String)
```

Comment:
Yeah, I think location should be a Place nearly everywhere (except near the ES interface).
I started down that road, but gave up.